cmd/agent/app/run.go

```diff
@@ -221,6 +219,11 @@ func StartAgent() error {
 		return fmt.Errorf("Error while setting up logging, exiting: %v", err)
 	}
 
+	if errStart != nil {
```

Comment:
I would move this `if` before the `Error while setting up logging` previous `if`. For clarity, given things will be out of order, we might want to rename `err` to `loggerSetupErr` or something like that.